### PR TITLE
Tests to speed up docs 

### PR DESCRIPTION
**Pull request by Wauplin**
Related to https://github.com/huggingface/doc-builder/pull/487 and [internal slack thread](https://huggingface.slack.com/archives/C04F8N7FQNL/p1711384899462349?thread_ts=1711041424.720769&cid=C04F8N7FQNL). There is now a `custom_container` option when building docs in CI. When set to `""` (instead of `"huggingface/transformers-doc-builder"` by default), we don't run the CI inside a container, therefore saving ~2min of download time. The plan is to test disabling the transformers container on a few "big" repo and if everything works correctly, we will stop making it the default container. More details on https://github.com/huggingface/doc-builder/pull/487.

cc @mishig25

### CODE REVIEW

File: .github/workflows/build_pr_documentation.yaml [hunk 1]
```diff
@@ -9,9 +9,10 @@ concurrency:
...
 jobs:
   build:
-    uses: huggingface/doc-builder/.github/workflows/build_pr_documentation.yml@main
+    uses: huggingface/doc-builder/.github/workflows/build_pr_documentation.yml@test-build-pr-outside-container
     with:
       commit_sha: ${{ github.event.pull_request.head.sha }}
       pr_number: ${{ github.event.number }}
       package: huggingface_hub
       languages: cn de fr en hi ko
+      container_image: ""
```
Comment (mishig25) [inline .github/workflows/build_pr_documentation.yaml:18]: when it is empty string, does it gets treated as a supplied value OR does it gets treated as default value (which is transformers container) ?

Comment (Wauplin) [reply .github/workflows/build_pr_documentation.yaml:18]: This is what I am tested as it's not documented :smile:

Comment (Wauplin) [reply .github/workflows/build_pr_documentation.yaml:18]: But it looks like it's treated as an empty value, so container is not used at all: https://github.com/huggingface/huggingface_hub/actions/runs/8423002045/job/23063619212?pr=2140 (i.e. it just runs on the base ubuntu:latest).

**Comment by Wauplin** [comment]
Dropping this now that https://github.com/huggingface/doc-builder/pull/487 has been successfully merged :)

**Pull request closed without merging**